lib/daru/vector.rb

```diff
@@ -531,36 +533,42 @@ def all? &block
     #   v = Daru::Vector.new ["My first guitar", "jazz", "guitar"]
     #   # Say you want to sort these strings by length.
     #   v.sort(ascending: false) { |a,b| a.length <=> b.length }
-    def sort opts={}
-      opts = {
-        ascending: true
-      }.merge(opts)
-
-      vector_index = @data.each_with_index
-      vector_index =
-        if block_given?
-          vector_index.sort { |a,b| yield(a[0], b[0]) }
-        else
-          vector_index.sort { |(av, ai), (bv, bi)|
-            if !av.nil? && !bv.nil?
-              av <=> bv
-            elsif av.nil? && bv.nil?
-              ai <=> bi
-            elsif av.nil?
-              opts[:ascending] ? -1 : 1
-            else
-              opts[:ascending] ? 1 : -1
-            end
-          }
-        end
-      vector_index.reverse! unless opts[:ascending]
+    def sort opts={}, &block
+      opts = {ascending: true}.merge(opts)
+
+      vector_index = resort_index(@data.each_with_index, opts, &block)
       vector, index = vector_index.transpose
       old_index = @index.to_a
-      index = index.map { |i| old_index[i] }
+      index.map! { |i| old_index[i] }
 
       Daru::Vector.new(vector, index: index, name: @name, metadata: @metadata.dup, dtype: @dtype)
     end
 
+    DEFAULT_SORTER = lambda { |(av, ai), (bv, bi)|
+      # FIXME: I suspect it can be replaced with just
+      #   [av || -Float::INFINITY, ai] <=> [bv || -Float::INFINITY, bi]
+      # but need more thorough check
```

Comment:
`[av || -Float::INFINITY, ai] <=> [bv || -Float::INFINITY, bi]` would fail I think when `av` is something other than integer, say String and `bv` is nil.